src/lib.rs

```diff
@@ -227,52 +450,62 @@ impl Graph {
         Ok(())
     }
 
-    pub fn port_name(&self, port: PortRef) -> Result<&'_ str, Error> {
+    pub fn port_ident(&self, port: PortRef) -> Result<&'_ PortIdent<P>, Error> {
         self.port_check(port)?;
-        Ok(&self.port_names[port.0])
+        Ok(&self.port_identifiers[port.0])
     }
 
-    pub fn node_name(&self, node: NodeRef) -> Result<&'_ str, Error> {
+    pub fn node_ident(&self, node: NodeRef) -> Result<&'_ NodeIdent<N, PT>, Error> {
         self.node_check(node)?;
-        Ok(&self.node_names[node.0])
-    }
-
-    fn node_count(&self) -> usize {
-        self.ports.len()
-    }
-
-    fn port_count(&self) -> usize {
-        self.port_data.len()
+        Ok(&self.node_identifiers[node.0])
     }
 
-    fn node_check(&self, n: NodeRef) -> Result<(), Error> {
+    pub fn node_check(&self, n: NodeRef) -> Result<(), Error> {
```

Comment:
Does this need to be public?

Reply:
I'm using this in Meadowlark to test if a node exists. I'm using the`NodeRef` as the identifiers for my nodes.

Reply:
I'm using the same `index into Vec / Vec of free_nodes` system in Meadowlark. I *could* change it to use a HashMap in Meadowlark, but since this system is already here, I figured I might as well use it.